Complete missing Flow declarations in URL

mischnic opened the pull request:
## Summary

When transpiling flow with a Babel config like
```
{
	plugins:  [['@babel/plugin-transform-flow-strip-types', {requireDirective: true}]]
}
```
all files containing Flow syntax need to have `@flow` at the top of the file.
```
node_modules/react-native/Libraries/Blob/URL.js: A @flow directive is required when using Flow annotations with the `requireDirective` option.
  55 |   _searchParams = [];
  56 |
> 57 |   constructor(params: any) {
     |                     ^^^^^
  58 |     if (typeof params === 'object') {
  59 |       Object.keys(params).forEach(key => this.append(key, params[key]));
  60 |     }
```

In URL, it was removed (mistakenly I think) by https://github.com/facebook/react-native/commit/63038500a23d2dc10798cc116bb70c03064db8bc#diff-552f9731d5c9bc329105a5f4224319966395e59b5bb23202b756c5d152e0f007. Only the `strict-local´ part should have been removed, not the whole line.

## Changelog

<!-- Help reviewers and the release process by writing your own changelog entry. For an example, see:
https://github.com/facebook/react-native/wiki/Changelog
-->

[General] [Fixed] - Complete missing Flow declarations in URL

## Test Plan

See above

lunaleaps commented:
@mischnic It looks like this was also called out in the original PR: https://github.com/facebook/react-native/pull/22901/files#r246114640 but I guess if flow passes here then it's not an issue?

mischnic commented:
>  then it's not an issue?

Flow declarations without `@flow` are essentially dead code (and you can run into problems with other tools such as Babel or flow-remove-types).

Up until now I didn't actually test if Flow passed with my changes, so now I've:
- removed the two "stray" flow type declarations in WebSocketInterceptor, because there are many missing types in that file
- added all missing types in URL, so Flow passes now

lunaleaps commented on Libraries/WebSocket/WebSocketInterceptor.js:
```diff
@@ -17,8 +17,8 @@ const originalRCTWebSocketSend = NativeWebSocketModule.send;
 
-let eventEmitter: NativeEventEmitter;
-let subscriptions: Array<EventSubscription>;
+let eventEmitter;
```
Why was `@flow` removed from your last commit?

lunaleaps replied:
Could we leave this file untouched -- I realize the types could be dead but I'd prefer we do this in a separate PR so the intent is clearer

mischnic replied:
> a separate PR

Done https://github.com/facebook/react-native/pull/32570
